spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/support/SpringMvcContract.java

```diff
@@ -144,6 +154,25 @@ public MethodMetadata parseAndValidateMetadata(Class<?> targetType, Method metho
 		return md;
 	}
 
+	protected void parseRegexAndUpdatePath(MethodMetadata md, Method method) {
+		if (isAnnotated(method, RequestMapping.class)) {
+			String url = md.template().url();
+			int paramCount = method.getParameterTypes().length;
+			for (int i = 0; i < paramCount; i++) {
+				MethodParameter param = new SynthesizingMethodParameter(method, i);
+				param.initParameterNameDiscovery(parameterNameDiscoverer);
+				if (param.hasParameterAnnotation(PathVariable.class)) {
+					PathVariable pathVariable = param.getParameterAnnotation(PathVariable.class);
+					String varName = pathVariable.value();
+					url = url.replaceAll("\\{" + varName + ":.+?\\}", "{" + varName + "}");
+				}
+			}
+			Field urlField = ReflectionUtils.findField(RequestTemplate.class, "url");
```

Comment:
This worries me a bit. Without having looked myself, is there a way to  do this without reflection?